src/backend/tests/teams.test.ts

```diff
@@ -117,4 +117,40 @@ describe('Teams', () => {
       expect(prisma.team.findUnique).toHaveBeenCalledTimes(1);
     });
   });
+
+  describe('setTeamHead', () => {
+    test('setTeamHead head not found', async () => {
+      jest.spyOn(prisma.team, 'findUnique').mockResolvedValue(prismaTeam1);
```

Comment:
you have to mock the user.findUnique as well

Reply:
@Peyton-McKee Good point, but I found out that mocking the user.findUnique here will not trigger `NotFoundException `error on line 137 in teams.services.ts. 

In other words, mocking the user.findUnique won't make `await expect(callSetTeamHead)` to reject on line 129 in this file.

Reply:
or in other way, we can actually mock the user.findUnique by setting mockResolvedValue as null. I had to do that for team.findUnique in a team not found case or otherwise team.findUnique always returns sharedTeam1 obj (prob due to line 13 in test file). Lmk ur thoughts.

Reply:
Nvm adding mock resolves github tests as you mentioned above. Good to review again 💯